encode.go

```diff
@@ -0,0 +1,62 @@
+package gpp
+
+import (
+	"fmt"
+	"github.com/prebid/go-gpp/constants"
+	"github.com/prebid/go-gpp/util"
+	"sort"
+	"strings"
+)
+
+const (
+	// the first 6 bits of the header must always evaluate to the interger '3'.
+	gppType    byte = 0x3
+	gppVersion byte = 0x1
+)
+
+var (
+	duplicatedSectionErr = fmt.Errorf("duplicated sections")
+)
+
+func Encode(sections []Section) (string, error) {
+	bs := util.NewBitStream(nil)
```

Comment:
It may be more efficient to estimate the final size of the bitstream, and then create a `[]byte` with that capacity. That will save a lot of reallocations and copies as the bitstream is expanded with append statements. Most applications are not so memory constrained that allocating a handful of bytes more than needed per GPP string should be an issue.